simulator/modules/decode/bypass/data_bypass_interface.h

```diff
@@ -58,7 +60,11 @@ class BypassCommand
 
     // returns an index of the port where bypassed data should be get from
     // in accordance with passed bypass command
-    auto get_bypass_direction() const { return static_cast<uint8>( get_bypassing_stage()); }
+    auto get_bypass_direction() const
+    {
+        const auto bypass_direction = static_cast<uint8>( bypassing_stage);
+        return ( bypassing_stage == 0_RSG) ? bypass_direction : bypass_direction - 1; 
```

Comment:
Why?

Reply:
Please put a comment then.

However I still do not understand that — how is 0_RSG connected to EXECUTE_1?

Reply:
EXECUTE_0 is 0_RSG and bypass_direction for EXECUTE_0 should be equal to 0.
EXECUTE_1 is 1_RSG and we cannot bypass data from it.
EXECUTE_2 is 2_RSG but we want to have its bypass_direction equal to 1.

Reply:
> EXECUTE_1 is 1_RSG and we cannot bypass data from it.

But the function returns 0, likewise for EXECUTE_0.